### DISCUSSION TITLE
Flashed Tuya TYJW2S-5V board but won't boot

### DISCUSSION THREAD

**Discussion by gooman-uk** [arendst/Tasmota#10464] [Q&A] [answered]
I have a panel heater that has Tuya built-in. Very similar to this device: https://templates.blakadder.com/klarstein_10034390.html

Except mine uses a Tuya TYJW2S-5V board (see https://developer.tuya.com/en/docs/iot/device-development/module/wifi-module/tyjw-series-module/wifijw2s5vmodule?id=K9605srhjahvz for specs).

I've removed the board from the heater and disconnected the MCU.

I've soldered header pins to pins 5-10 as described in the specs (3.3V, GND, GPIO0 (called BOOT here), Tx (using TXD0) and Rx).

I've used Tasmotizer to backup the firmware (successfully) and to erase and flash Tasmota 9.2. It connects and seems to flash OK.

But after a power cycle there's no Wifi AP. If I connect using Putty, there's nothing being output on the serial console.

I've tried flashing tasmota-minimal, but get the same response.

I've also tried flashing an earlier binary version I have working on some Sonoff devices (8.4.0).

Any guesses?

**Comment by sfromis** [comment]
Flashing tasmota-minimal.bin as the first binary bricks the device. Start with tasmota.bin or tasmota-lite.bin.

**Comment by Jason2866** [comment]
**Never** flash minimal as first firmware. This firmware is just and only for updating devices already running Tasmota. Flash Tasmota, if flashing is successful assemble the device and install it.
Many devices do not work if powered only with 3.3V from flasher.

**Reply by gooman-uk** [reply]
To be clear - I tried tasmota.bin first, which didn't work. 

I then assembled the device and tried it from its regular power supply - still no AP advertised.

**Reply by gooman-uk** [reply]
Just tried again. This time, when Tasmotizer prompted to power cycle the device, I unplugged it from the FTDI and reassembled the heater.

Then powered on - et voila, a Tasmota AP. Seems the first boot after flashing has to be in situ, or at least at 5V.

**Comment by gooman-uk** [comment, marked as answer]
As noted above - Seems the first boot after flashing has to be in situ, or at least at 5V.

**Comment by Frogmore42** [comment]
The Tuya device you have always requires 5V and has 5V TTL UART lines.  This means that you need to use a flasher that is set for 5V on the data lines (it might work with one set to 3V3, but it also might not).  The flasher also needs to provide 5V to the device while flashing it, since there is 3V3 regulator (and providing 3V3 to a 3V3 regulator will provide less than 3V3 out).

Note that this is very different than most esp8266 devices.  I have a Adafruit Huzzah that is similar.  It has a 6-pin header with a standard FTDI layout.  I have an actual FTDI made cable (manufactured by FTDI and not a cheap clone).  These cables provide VBUS (5V from USB), but come with different IO levels for the UART (I have 3V3 and 1V8 versions).  Some of the clones use the LDO in the FTDI (or clone) chip to provide 3V3 out, but the LDO is only rated for about 50mA which is not enough in many cases.  Other programmers have a proper regulator that can provide enough current to power the esp8266.  Before I noticed that my UART cable was actually providing 5V (and fixed it by adding a regulator) I discovered that I could make the relays work while powered by the UART cable.  I also think I destroyed one of the esp8266 on one of my Sonoff devices, since they are not rated to handle 5V.

**Reply by gooman-uk** [reply]
Thanks for the reply!

It certainly needs 5V to boot successfully (as I discovered when flashing in Tasmota). But the spec sheet for the board (https://developer.tuya.com/en/docs/iot/device-development/module/wifi-module/tyjw-series-module/wifijw2s5vmodule?id=K9605srhjahvz) seems to suggest that you should use what it calls the "Test Points" for flashing (pins 5-10).  I've soldered header pins to these points and successfully flashed Tasmota that way. I then verified that had worked by then using pins 1-4 with 5V and booted Tasmota successfully.

I then reflashed the backup using the same method, but it's unable to communicate with the MCU.

Reading more about esptool options, I'm wondering if it would be better to use that for the recovery, using the "-fm dout" option?
